Added recursion to the drush command

jyobb opened the pull request:
I have added a the ability to recursively send subcollections and books to archivematica. I also added some reporting.

jyobb committed:
removed slash from status verb

jyobb committed:
cleaned up debug code for pull request

jyobb committed:
added subdirectory ingest

jyobb committed:
added subdirectories

jyobb committed:
Added option for children

jyobb committed:
Cleaned up code documentation, better output logging, added to README

jyobb committed:
fixed messaging

jyobb committed:
fixed count and children flag

jyobb committed:
fixed 0 size aip error

jyobb committed:
Cleaned out testing code

Commit:
Added option to select itql

Commit:
Added sparql checkbox to admin form

Commit:
Adding sparql config option and first crack at query. With drupal message

Commit:
Added sparql query and debug code

Commit:
Debuging message added

Commit:
Debug test 2

Commit:
Variable setting test

Commit:
Added default value to use sparql option

Commit:
Testing sparql variable select

Commit:
Moved query selector into proper place in code for testing

Commit:
sparql edit

manez commented:
@jyobb we'd like to see these kinds of updates in Islandora Labs run much more smoothly (and quickly!) so we're working on changes to the merging workflow to open it up. Your comments very welcome, since you've been affected by this: 

https://groups.google.com/d/msg/islandora/BQzqUpEssAE/WwrllTgJFgAJ

DiegoPino commented on includes/utilities.inc:
```diff
@@ -91,7 +109,12 @@ function archidora_should_auto_enqueue($pid) {
+    }else{
+        $results = $ri->itqlQuery($current_query);
+    }
+var_dump($results);
```
var_dump needs to be removed here

DiegoPino replied:
@jyobb hi, still missing a few comments. Can you remove the var_dump?

DiegoPino commented on includes/archivematica.inc:
```diff
@@ -85,6 +85,9 @@ function archidora_send_to_archivematica(AbstractObject $object, $aip = NULL) {
       $aip->size += $object['OBJ']->size;
     }
     if ($aip->created) {
+
```
Coding style here looks strange. Probably removing those extra lines and identing the controls structure would help

DiegoPino commented on includes/admin.form.inc:
```diff
@@ -206,6 +206,12 @@ function archidora_admin_form($form, &$form_state) {
+   $form['archidora_use_sparql'] = array(
+       '#type' => 'checkbox',
+       '#title' => t('Check to enable SPARQL'),
+       '#description' => t('Leave unchecked for Mulgara triplestore'),
```
Mulgara uses SPARQL, maybe not 1.1 100% compliant (probable more than less than that). So the SPARQL that this module uses does not work in Mulgara?

jyobb replied:
I think I messed up with Git yet again.  I did not think that this code was in the pull request.  We are in the process of switching triplestores from mulgara to blazegraph.  The module was originally written with only a itql query.  I added this option to support blazegraph because I was uncertain of mulgara's sparql support.  I'll test it out in a mulgara environment and if it works I'll remove the itql altogether.

DiegoPino commented on archidora.drush.inc:
```diff
@@ -96,22 +105,98 @@ EOQ;
+    $relationships = $fedoraObject->relationships->get('info:fedora/fedora-system:def/model#', 'hasModel');
+    if(in_array($relationships[0]['object']['value'], $collections)){
+       //This is where you need to recurse to get the rest.
+       file_put_contents('/var/www/html/drupal/sites/all/modules/archidora/drush.log', "Processing child collection: " . $result['pid']['value'] ."\n", FILE_APPEND);
```
This is opinionanted. Not all people have drupal in that path. I would suggest using stream wrappers for this that resolve to real installation paths

DiegoPino commented on archidora.drush.inc:
```diff
@@ -96,22 +105,98 @@ EOQ;
+
+  foreach($pidArr as $k => $v){
+
+    $object = islandora_object_load($v);
```
this needs a "try" to make sure the object loads before sending to archivematica

DiegoPino commented on archidora.drush.inc:
```diff
@@ -96,22 +105,98 @@ EOQ;
+    }else{
+       array_push($pidArr, $result['pid']['value']);
+       $curPid = $result['pid']['value'];
+       file_put_contents('/var/www/html/drupal/sites/all/modules/archidora/drush.log', "Adding: " . $curPid ."\n", FILE_APPEND);
```
Same as further down. Hard coding a path is not a good idea. Using stream wrapper drupal can resolve is better

DiegoPino commented on archidora.drush.inc:
```diff
@@ -70,7 +75,11 @@ function archidora_send_collection_create_batch($target) {
-function archidora_send_collection_batch_operation($target, &$context) {
+function archidora_send_collection_batch_operation($target, $children, &$context) {
+
+  file_put_contents('/var/www/html/drupal/sites/all/modules/archidora/drush.log', "Starting new batch: " . $target ." with children set to " . $children . "\n");
```
Sames as further. No path hardcoding

DiegoPino commented on archidora.drush.inc:
```diff
@@ -17,13 +17,17 @@ function archidora_drush_command() {
     ),
     'examples' => array(
-      'drush -u 1 archidora-send-collection-to-archivematica',
+      'drush -u 1 archidora-send-collection-to-archivematica --target=pid --children=(yes,no)',
```
What happens with shared collections? how are cycles broken?

jyobb replied:
Cycles are only broken when there are no more collections. I'll look into shared collections.

DiegoPino commented:
@jyobb thanks for this. It would be useful to get some help on testing this, the pull request s farther large and i don't know were to look at exactly. I added some notes (not requiring changes, never liked that anyway), look at them more as suggested changes.

jyobb commented:
Melissa, I will add my feedback.

Diego,

Thanks for taking a look at this and I apologise for the state of the code
I thought I had cleaned up the code better than that.  I have another
branch where I have added sparql support because we are moving all our
instances to blazegraph.  What is the best path to making these changes? Do
I do another pull request?

On Wed, Apr 12, 2017 at 9:33 AM, Diego Pino Navarro <
notifications@github.com> wrote:

> @jyobb <https://github.com/jyobb> thanks for this. It would be useful to
> get some help on testing this, the pull request s farther large and i don't
> know were to look at exactly. I added some notes (not requiring changes,
> never liked that anyway), look at them more as suggested changes.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Islandora-Labs/archidora/pull/23#issuecomment-293616897>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACqS9jarriHwUO8wmJbyZ9EMFj3L_ndQks5rvO7GgaJpZM4K0bVy>
> .
>

DiegoPino commented:
@jyobb cool.! You can still work on this pull request. Just merge your other changes and cleanup into jyobb:7.x and that will update this one. Or are the changes completely different/divergent to this one?  In that case you can simply force push that other branch into/replacing jyobb:7.x  which in turn will update this pull. Cheers!

Commit:
Removed hard coded paths. Added check to ensure object is loaded before pushing to archivematica.

DiegoPino commented:
@jyobb looks much better! Will give it a try tomorrow. Please please ping me if I forget about this, sometimes I just forget stuff. Thanks!

Commit:
Fixed drush.log path

DiegoPino commented on includes/utilities.inc:
```diff
@@ -66,6 +75,15 @@ function archidora_should_auto_enqueue($pid) {
+
+  $sparql_enabled = variable_get('archidora_use_sparql');
+
+  $sparql_query = 'SELECT ?disabled FROM  <info:edu.si.fedora#ri> WHERE {'.
```
Hi, just curious, why the <info:edu.si.fedora#ri> instead of the  shortcut <#ri> ?

DiegoPino commented on includes/utilities.inc:
```diff
@@ -66,6 +75,15 @@ function archidora_should_auto_enqueue($pid) {
   if (module_exists('islandora_compound_object')) {
     $predicates[] = variable_get('islandora_compound_object_relationship', 'isConstituentOf');
```
What about other relationships, books, newspapers. Is this just for compounds?

DiegoPino reviewed (commented):
Hey, to keep this running, i added a few questions. This is labs and there is a new workflow (more free, less diegos asking for stuff!) but i guess its ok to keep talking =)

Commit:
catching up

jyobb committed:
sparql fix

Commit:
Merge branch '7.x' of github.com:jyobb/archidora into 7.x
